tutorials/surrogate/cifar10_example.py

```diff
@@ -0,0 +1,429 @@
+        "CAB": Permutation.CAB,
+        "CBA": Permutation.CBA,
+    }
+
+    permutation = permutations[perm]  # Get permutation.
+
+    num_classes: int = 10  # Number of classes in CIFAR10 dataset.
+    in_channels: int = 3  # Number of channels in CIFAR10 images.
+
+    loss_fn = (
+        torch.nn.CrossEntropyLoss()
+    )  # Use cross-entropy loss for multi-class classification.
+
+    # use non null weight decay
+    weight_decay = 1e-4
+    grad_clip = 0.1
+
+    model = Net(
+        in_channels, num_classes, permutation, lr, loss_fn, weight_decay
+    ).to(device)  # Set up neural network with specified hyperparameters.
+    model.best_accuracy = 0.0  # Initialize the model's best validation accuracy.
+
+    train_loader, val_loader = get_data_loaders(
+        batch_size=8
+    )  # Get training and validation data loaders.
+
+    # Configure optimizer
+    optimizer = model.configure_optimizers()
+
+    # init avg_val_loss
+    avg_val_loss: float = 0.0
+
+    for epoch in range(epochs):
+        model.train()
+        total_train_loss = 0
+        # Training loop
+        for batch_idx, (data, target) in enumerate(train_loader):
+            data, target = data.to(device), target.to(device)
+            # zero parameter gradients
+            optimizer.zero_grad()
+            # forward + backward + optimize
+            loss = model.training_step((data, target), batch_idx)
+            loss.backward()
+
+            # use gradient clipping
+            nn.utils.clip_grad_value_(model.parameters(), grad_clip)
+
+            optimizer.step()
+            # update loss
+            total_train_loss += loss.item()
+
+        avg_train_loss = total_train_loss / len(train_loader)
+        print(f"Epoch {epoch+1}: Avg Training Loss: {avg_train_loss}")
+
+        # Validation loop
+        model.eval()
+        total_val_loss = 0
+        with torch.no_grad():
+            for batch_idx, (data, target) in enumerate(val_loader):
+                data, target = data.to(device), target.to(device)
+                # forward
+                loss = model.validation_step((data, target), batch_idx)
+                # update loss
+                total_val_loss += loss.item()
+
+        avg_val_loss = total_val_loss / len(val_loader)
+        print(f"Epoch {epoch+1}: Avg Validation Loss: {avg_val_loss}")
+
+        yield avg_val_loss
+
+
+def set_seeds(seed_value=42):
+    """
+    Set seed for reproducibility.
+    """
+    random.seed(seed_value)  # Python random module
+    torch.manual_seed(seed_value)  # pytorch random number generator for CPU
+    torch.cuda.manual_seed(seed_value)  # pytorch random number generator for all GPUs
+    torch.cuda.manual_seed_all(seed_value)  # for multi-GPU.
+    torch.backends.cudnn.deterministic = True  # use deterministic algorithms.
+    torch.backends.cudnn.benchmark = False  # disable to be deterministic.
+    os.environ['PYTHONHASHSEED'] = str(seed_value)  # python hash seed.
```

Comment:
See comments in other file. Maybe, this could be refactored into a utility function?

Reply:
You mean moving the function to `utils.py`?

Reply:
Yes, exactly. Might be handy to have such a function for other use cases as well.

Reply:
This would add PyTorch as dependency though. If this is not wanted I could make a reduced version with only the `random`, `np.random` and `PYTHONHASHSEED` seed set.

Alternatively something like this
```Python
try:
    import torch
except ImportError:
    torch = None
```

I don't think adding PyTorch to Propulates `requirements.txt` would be a good idea. It's a large library and there are many variants.